app/src/main/java/eu/darken/capod/monitor/ui/MonitorNotifications.kt

```diff
@@ -80,31 +79,30 @@ class MonitorNotifications @Inject constructor(
             // Options here should be mutually exclusive, and are prioritized by their order of importance
             // Some options are omitted here, as they will conflict with other options
             // TODO: Implement a settings pane to allow user to customize this
-            device.apply {
+            when (device) {
```

Comment:
Nice variant! My style would have been something like this, but everyone writes their code a little different :heart: 

```kotlin
@StringRes val contentTitleRes = when {
                // Pods charging state
                // This goes first as pods should not be worn if it is still charging
                device is HasChargeDetection && device.isHeadsetBeingCharged -> {
                    eu.darken.capod.common.R.string.pods_charging_label
                }

                // Pods wear state
                device is HasEarDetection -> when {
                    device.isBeingWorn -> eu.darken.capod.common.R.string.headset_being_worn_label
                    else -> eu.darken.capod.common.R.string.headset_not_being_worn_label
                }

                // Case charge state
                // This is under pods wear state as we don't want it conflicting with it
                device is HasCase && device.isCaseCharging -> {
                    eu.darken.capod.common.R.string.pods_charging_label
                }
                
                else -> eu.darken.capod.common.R.string.pods_case_unknown_state
            }
```

Reply:
Seems like I am still not accustomed to the Kotlin syntax 😅